src/main/scala/sangria/schema/Schema.scala

```diff
@@ -295,6 +299,9 @@ case class UnionType[Ctx](
     astNodes: Vector[ast.AstNode] = Vector.empty) extends OutputType[Any] with CompositeType[Any] with AbstractType with NullableType with UnmodifiedType with HasAstInfo {
   def rename(newName: String) = copy(name = newName).asInstanceOf[this.type]
   def toAst: ast.TypeDefinition = SchemaRenderer.renderType(this)
+  def map[T](func: T => Any): OutputType[T] with MappedAbstractType[T] = new UnionType[Ctx](name, description, types, astDirectives, astNodes) with MappedAbstractType[T] {
+    override def contraMap(value: T): Any = func(value)
+  }.asInstanceOf[OutputType[T] with MappedAbstractType[T]]
```

Comment:
Not 100% sure how I feel about this part. On one hand, I can understand practical reasons behind it, but on the other hand I'm not a big fan of such structures especially for case classes.

Though it provides quite non intrusive way to enable this feature. I think this is nice, especially considering that it not yet clear how scala 3 union types will affect the design.

Reply:
Yeah so you nailed my thought process here, I did not way to break the UnionType contract. Honestly, I just wasn't sure what you'd think, so i'd be looking for your advice on this.  I also dislike doing this to case classes.